src/libUtils/EvmUtils.cpp

```diff
@@ -0,0 +1,182 @@
+    arr_ret.append(params.m_code);
+  }
+  arr_ret.append(params.m_data);
+  arr_ret.append(params.m_apparent_value.str());
+  arr_ret.append(Json::Value::UInt64(params.m_available_gas));
+
+  if (LOG_SC) {
+    LOG_GENERAL(WARNING, "Sending to EVM-DS" << arr_ret);
+  }
+
+  return arr_ret;
+}
+
+bool EvmUtils::EvmUpdateContractStateAndAccount(
+    Account* contractAccount, evmproj::ApplyInstructions& op) {
+  if (op.OperationType() == "modify") {
+    // Reset State for this contract
+    try {
+      if (op.isResetStorage()) {
+        //
+        // Reset Meta Data for this Address effectively clears down the contract
+        // storage for this Contract
+        //
+        std::map<std::string, bytes> t_metadata;
+
+        t_metadata.emplace(
+            Contract::ContractStorage::GenerateStorageKey(
+                Address(op.Address()), CONTRACT_ADDR_INDICATOR, {}),
+            Address(op.Address()).asBytes());
+
+        if (!contractAccount->UpdateStates(Address(op.Address()), t_metadata,
+                                           {}, true)) {
+          LOG_GENERAL(WARNING,
+                      "Account::UpdateStates reset metaData and Merkyle tree");
+        }
+        contractAccount->SetStorageRoot(dev::h256());
+      }
+    } catch (std::exception& e) {
+      // for now catch any generic exceptions and report them
+      // will exmine exact possibilities and catch specific exceptions.
+      LOG_GENERAL(WARNING,
+                  "Exception thrown trying to reset storage " << e.what());
+    }
+    // If Instructed to reset the Code do so and call SetImmutable to reset
+    // the hash
+    try {
+      if (op.Code().size() > 0)
+        contractAccount->SetImmutable(
+            DataConversion::StringToCharArray("EVM" + op.Code()),
+            contractAccount->GetInitData());
+    } catch (std::exception& e) {
+      // for now catch any generic exceptions and report them
+      // will exmine exact possibilities and catch specific exceptions.
+      LOG_GENERAL(WARNING, "Exception thrown trying to update Contract code "
+                               << e.what());
+    }
+    // Actually Update the state for the contract
+    try {
+      for (const auto& it : op.Storage()) {
+        if (!Contract::ContractStorage::GetContractStorage().UpdateStateValue(
+                Address(op.Address()),
+                DataConversion::StringToCharArray(it.Key()), 0,
+                DataConversion::StringToCharArray(it.Value()), 0)) {
+          return false;
+        }
+      }
+    } catch (std::exception& e) {
+      // for now catch any generic exceptions and report them
+      // will exmine exact possibilities and catch specific exceptions.
+      LOG_GENERAL(WARNING,
+                  "Exception thrown trying to update state on the contract "
+                      << e.what());
+    }
+    try {
+      if (op.Balance().size())
+        contractAccount->SetBalance(uint128_t(op.Balance()));
```

Comment:
the instruction has the address, why are we setting the contract account's balance?